File: t/VB/MedicalViewerDemo/UI/AdaptiveContrastDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports Leadtools.ImageProcessing.Color

Imports Leadtools.MedicalViewer
Imports Leadtools

Namespace MedicalViewerDemo
   Partial Public Class AdaptiveContrastDialog : Inherits Form
      Private _firstTime As Boolean
      Private _originalBitmap As RasterImage
      Private _mainForm As MainForm

      Public Sub New(ByVal mainForm As MainForm)
         _mainForm = mainForm
         InitializeComponent()
         _cbType.SelectedIndex = 0
         _firstTime = True
      End Sub

      Private Sub _num_Leave(ByVal sender As Object, ByVal e As System.EventArgs)
         DialogUtilities.NumericOnLeave(sender)
      End Sub

      Private Sub _btnOk_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles _btnOk.Click
         _mainForm.FilterOk_Click(_originalBitmap, _firstTime, AddressOf ApplyFilter)
      End Sub

      Private Sub _btnApply_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnApply.Click
         _mainForm.FilterApply_Click(_firstTime, _originalBitmap, AddressOf ApplyFilter)
      End Sub

      Private Sub _btnCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnCancel.Click
         _mainForm.FilterCancel_Click(_firstTime, _originalBitmap, True)
      End Sub

      Private Sub ApplyFilter()
         Dim type As AdaptiveContrastCommandType = AdaptiveContrastCommandType.Exponential
         Select Case _cbType.SelectedIndex
            Case 0
               type = AdaptiveContrastCommandType.Exponential
            Case 1
               type = AdaptiveContrastCommandType.Linear
         End Select
         Dim command As AdaptiveContrastCommand = New AdaptiveContrastCommand(Convert.ToInt32(_numSize.Value), Convert.ToInt32(_numAmount.Value), type)
         _mainForm.FilterRunCommand(command, True)
      End Sub

      Private Sub AdaptiveContrastDialog_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs) Handles Me.FormClosing
         If Not _originalBitmap Is Nothing Then
            _originalBitmap.Dispose()
         End If
      End Sub
   End Class
End Namespace

File: t/CS/MedicalViewerDemo/UI/AdaptiveContrastDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Leadtools.ImageProcessing.Color;

using Leadtools.Demos;
using Leadtools.MedicalViewer;
using Leadtools;

namespace MedicalViewerDemo
{
    public partial class AdaptiveContrastDialog : Form
    {
        private bool _firstTime;
        private RasterImage _originalBitmap;
        private MainForm _mainForm;

        public AdaptiveContrastDialog(MainForm mainForm)
        {
            _mainForm = mainForm;
            InitializeComponent();
            _cbType.SelectedIndex = 0;
            _firstTime = true;
        }

        private void _num_Leave(object sender, System.EventArgs e)
        {
            DialogUtilities.NumericOnLeave(sender);
        }

        private void _btnOk_Click(object sender, System.EventArgs e)
        {
           _mainForm.FilterOk_Click(_originalBitmap, _firstTime, ApplyFilter);
        }

        private void _btnApply_Click(object sender, EventArgs e)
        {
           _mainForm.FilterApply_Click(ref _firstTime, ref _originalBitmap, ApplyFilter); 
        }

        private void _btnCancel_Click(object sender, EventArgs e)
        {
           _mainForm.FilterCancel_Click(_firstTime, _originalBitmap, true);
        }

        private void ApplyFilter()
        {
            AdaptiveContrastCommandType type = AdaptiveContrastCommandType.Exponential;
            switch (_cbType.SelectedIndex)
            {
                case 0:
                    type = AdaptiveContrastCommandType.Exponential;
                    break;
                case 1:
                    type = AdaptiveContrastCommandType.Linear;
                    break;
            }
            AdaptiveContrastCommand command = new AdaptiveContrastCommand(Convert.ToInt32(_numSize.Value), Convert.ToInt32(_numAmount.Value), type);
            _mainForm.FilterRunCommand(command, true, false);
        }

        private void AdaptiveContrastDialog_FormClosing(object sender, FormClosingEventArgs e)
        {
           if (_originalBitmap != null)
              _originalBitmap.Dispose();
        }
    }
}
